src/include/gurt/types.h

```diff
@@ -97,6 +97,20 @@ typedef struct {
 	d_iov_t		*sg_iovs;
 } d_sg_list_t;
 
+/**
+ * c string buffer
+ */
+struct d_string_buffer_t {
+	/** c string status */
+	unsigned int status;
```

Comment:
Does the status actually help anything?  It would mean checking the return code of every call but that doesn't seem particularly onerous.

Reply:
There are two options/use cases:

* Check the return code on every call, for instance for small strings that require a single call to be created.

* Does not check the return code, use as many calls on the same string buffer as required. Then, check the status of the string buffer. I added a unit test to demo it. If the `d_string_buffer_t` is dirty, `d_write_string_buffer()` returns immediately